templates/lxc-local.in

```diff
@@ -0,0 +1,242 @@
+#!/bin/sh
+
+# Client script for LXC container images.
+#
+# Copyright © 2018 Stéphane Graber <stgraber@ubuntu.com>
+# Copyright © 2018 Christian Brauner <christian.brauner@ubuntu.com>
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 2.1 of the License, or (at your option) any later version.
+
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+#  USA
+
+set -eu
+
+LXC_HOOK_DIR="@LXCHOOKDIR@"
+LXC_TEMPLATE_CONFIG="@LXCTEMPLATECONFIG@"
+
+LXC_NAME=
+LXC_PATH=
+LXC_ROOTFS=
+LXC_CONFIG=
+
+# Detect use under userns (unsupported)
+for arg in "$@"; do
+  [ "$arg" = "--" ] && break
+  if [ "$arg" = "--mapped-uid" ] || [ "$arg" = "--mapped-gid" ]; then
+    echo "This template can't be used for unprivileged containers." 1>&2
+    echo "You may want to try the \"download\" template instead." 1>&2
```

Comment:
Hmm, why? We're only unpacking tarballs, which should be fine unprivileged.

Reply:
lxc-usernsexec currently unpacks in a new user namespace. Do we care about device nodes not being created?

Reply:
Nope, we don't and we in fact have logic in the download template specifically for this. There's a file in the metadata tarball which lists all the device nodes, those can then be excluded when unpacking the tarball.

Reply:
Added `relevant_file` function.